akka-remote/src/main/scala/akka/remote/RemoteWatcher.scala

```diff
@@ -287,15 +287,9 @@ private[akka] class RemoteWatcher(
    * does not exist.
    */
   def reWatch(address: Address): Unit =
```

Comment:
The rewatch mechanism was introduced in 637598a28bc5aafaca27b801f442cf9b20009807
When actor system was restarted quickly the new system replied to heartbeats and Terminated was never triggered for actors in old system.

Is that really solved by this?

The test fails when I remove the reWatch: akka.remote.RemoteNodeShutdownAndComesBack